python/tidy/tidy.py

```diff
@@ -356,19 +356,16 @@ def check_by_line(file_name: str, lines: list[bytes]):
             yield error
 
 
-def check_flake8(file_name, contents):
+def check_ruff_lints(file_name, contents):
     if not file_name.endswith(".py"):
         return
 
-    output = ""
     try:
-        args = ["flake8", file_name]
+        args = ["ruff", "check", "--output-format", "json", file_name]
         subprocess.check_output(args, universal_newlines=True)
```

Comment:
I think we should just run ruff without passing it any files, this way we ensure that test-tidy is compatible with standalone runs (and ruff exclusions are correct).

Reply:
Note that this will always lint all files, not just the modified ones.